internal/dcache/file_manager/worker.go

```diff
@@ -104,11 +105,21 @@ func (wp *workerPool) queueWork(file *DcacheFile, chunk *StagedChunk, get_chunk
 
 func (wp *workerPool) readChunk(task *task) {
 	log.Info("DistributedCache::readChunk : Reading chunk idx : %d, file: %s", task.chunk.Idx, task.file.FileMetadata.Filename)
-	var err error
-
 	// Read From the Dcache
-	//Call MvRead method for reading the chunk.
-	// err = rm.MVRead()
+	readMVReq := &rm.ReadMvRequest{
+		FileID:         task.file.FileMetadata.FileID,
+		MvName:         getMVForChunk(task.chunk, &task.file.FileMetadata.FileLayout),
+		ChunkIndex:     task.chunk.Idx,
+		OffsetInChunk:  task.chunk.Idx * task.file.FileMetadata.FileLayout.ChunkSize,
```

Comment:
this is offset within the chunk not within the file, so if you are reading full chunk, this should be 0.